src/Raven.Pal/inc/rvn_internal.h

```diff
@@ -0,0 +1,12 @@
+#include "rvn.h"
+
+typedef enum rvn_write_mode
+{
+    rvn_mode_default,
+    rvn_write_mode_file_io,
+    rvn_write_mode_io_ring,
+    rvn_write_mode_mmap,
+} rvn_write_mode;
+
```

Comment:
The same here, each one should have a proper documentation of the guarantees, mode of usage, what are their differences, etc.